ckb-bin/src/subcommand/run.rs

```diff
@@ -192,33 +193,38 @@ fn verify_genesis(shared: &Shared) -> Result<(), ExitCode> {
 
 fn check_spec(shared: &Shared, args: &RunArgs) -> Result<(), ExitCode> {
     let store = shared.store();
-    if let Some(spec_hash) = store.get_chain_spec_hash() {
-        if args.chain_spec_hash != spec_hash && !args.skip_chain_spec_check {
+    if args.override_chain_spec {
+        write_chain_spec_hash(store, &args.chain_spec_hash)?;
+    } else if store.get_chain_spec_hash().is_none() {
+        write_chain_spec_hash(store, &args.chain_spec_hash)?;
+    } else if !args.skip_chain_spec_check {
+        let stored_spec_hash = store.get_chain_spec_hash().expect("checked above");
+        if args.chain_spec_hash != stored_spec_hash {
             eprintln!(
-                "chain_spec_hash mismatch Config({}) storage({}), pass command line argument --skip-spec-check if you are sure that the two different chains are compatible.",
-                args.chain_spec_hash, spec_hash
+                "chain_spec_hash mismatch Config({}) storage({}), pass command line argument \
+                --skip-spec-check if you are sure that the two different chains are compatible; \
+                or pass --override-spec to force overriding stored chain spec with configured chain spec",
+                args.chain_spec_hash, stored_spec_hash
             );
             return Err(ExitCode::Config);
         }
-    } else {
-        store
-            .put_chain_spec_hash(&args.chain_spec_hash)
-            .map_err(|err| {
-                eprintln!(
-                    "Touch chain_spec_hash {} error: {}",
-                    args.chain_spec_hash, err
-                );
-                ExitCode::IO
-            })?;
-        info_target!(
-            crate::LOG_TARGET_MAIN,
-            "Touch chain spec hash: {}",
-            args.chain_spec_hash
-        );
     }
     Ok(())
 }
 
+fn write_chain_spec_hash(store: &ChainDB, chain_spec_hash: &Byte32) -> Result<(), ExitCode> {
```

Comment:
I think there should have a log to tell the user whether the chain spec hash was overwritten or not (`chain-spec-hash-old == chain-spec-hash-new`).

Reply:
An informational log is worthy.
And I am going to rename "override" to "overwrite" after reading your comments.